polygon/heimdall/reader.go

```diff
@@ -0,0 +1,57 @@
+package heimdall
+
+import (
+	"context"
+
+	libcommon "github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/log/v3"
+	"github.com/erigontech/erigon/polygon/bor/valset"
+)
+
+type Reader struct {
```

Comment:
should this be called ServiceReader?

Reply:
I like `heimdall.Reader` because it gives the sense that it is reading heimdall data, whereas `heimdall.ServiceReader` seems like it is reading from a "service" of some sort - a service which doesn't exist in the case of standalone rpcdaemon.

`heimdall.StoreReader` seems more appropriate if `heimdall.Reader` is too generic since that is what we are reading from here.

Reply:
Ok, probably a good idea to keep consistent with bridge.Reader for now. We can refine this in the future if needed. Btw note that the heimdall.Service and bridge.Service should have their own grpc server which the remote rpcdaemon will be hitting. So in a way im not sure if this is totally accurate - "service which doesn't exist in the case of standalone rpcdaemon."